src/modules/FileLocksmith/FileLocksmithLibInterop/NtdllExtensions.cpp

```diff
@@ -309,9 +309,7 @@ std::wstring NtdllExtensions::pid_to_user(DWORD pid)
     }
 
     DWORD token_size = 0;
-    GetTokenInformation(token, TokenUser, nullptr, 0, &token_size);
-
-    if (token_size < 0)
+    if (!GetTokenInformation(token, TokenUser, nullptr, 0, &token_size))
```

Comment:
Looking at the documentation for `GetTokenInformation`, this function looks like it was just trying to get the actual size needed for the next call with a proper buffer size. So this doesn't look like proper error checking, since that needs to take token_size into account as well.
https://learn.microsoft.com/en-us/windows/win32/api/securitybaseapi/nf-securitybaseapi-gettokeninformation
When we're just calling to get the required size of the buffer, does it count as a success or a fail?

Reply:
Good catch, thanks! Improved